src/main/java/com/sap/broker/budgie/resources/api/ServiceInstancesResource.java

```diff
@@ -38,69 +41,274 @@ public ServiceInstancesResource(ServiceBroker serviceBroker) {
+    }
+
+    private Response bindAsync(String configId, UUID bindingId, ServiceInstance serviceInstance, Binding binding) {
+        AsyncOperation asyncOperation = new AsyncOperation(() -> {
+            if (behaviorEngine.shouldOperationFail(configId, FailConfiguration.OperationType.BIND, serviceInstance).isPresent()) {
+                return new AsyncOperationState(AsyncOperationState.State.FAILED);
+            }
+            if (serviceInstance.getBinding(bindingId) != null) {
+                return new AsyncOperationState(AsyncOperationState.State.FAILED);
+            }
+            serviceInstance.bind(bindingId, binding);
+            return new AsyncOperationState(AsyncOperationState.State.SUCCEEDED);
+        }, behaviorEngine.getDuration(configId));
+        asyncOperationManager.addOperation(bindingId, asyncOperation);
+        return emptyBodyResponse(Status.ACCEPTED);
+    }
+
+    private Response unbindSync(String configId, UUID bindingId, ServiceInstance serviceInstance) {
+        sleepIfWanted(configId);
+        Optional<Integer> optionalStatusCode = behaviorEngine.shouldOperationFail(configId, FailConfiguration.OperationType.UNBIND, serviceInstance);
+        if (optionalStatusCode.isPresent()) {
+            return emptyBodyResponse(optionalStatusCode.get());
+        }
+        if (serviceInstance.getBinding(bindingId) == null) {
+            return emptyBodyResponse(Status.GONE);
+        }
+        serviceInstance.unbind(bindingId);
+        return emptyBodyResponse(Status.OK);
+    }
+
+    private Response unbindAsync(String configId, UUID bindingId, ServiceInstance serviceInstance) {
+        AsyncOperation asyncOperation = new AsyncOperation(() -> {
+            if (behaviorEngine.shouldOperationFail(configId, FailConfiguration.OperationType.UNBIND, serviceInstance).isPresent()) {
+                return new AsyncOperationState(AsyncOperationState.State.FAILED);
+            }
+            serviceInstance.unbind(bindingId);
+            return new AsyncOperationState(AsyncOperationState.State.SUCCEEDED);
+        }, behaviorEngine.getDuration(configId));
+        asyncOperationManager.addOperation(bindingId, asyncOperation);
+        return emptyBodyResponse(Status.ACCEPTED);
+    }
+
+    private void sleepIfWanted(String configId) {
+        try {
+            if (behaviorEngine.getDuration(configId) > 0) {
+                Thread.sleep(behaviorEngine.getDuration(configId));
+            }
+        } catch (InterruptedException e) {
+            throw new InterruptedOperationException(e);
         }
-        serviceBroker.delete(id);
-        return Response.ok()
-            .entity(new Object())
-            .build();
     }
 
+    private Response emptyBodyResponse(Status httpStatus) {
+        return emptyBodyResponse(httpStatus.getStatusCode());
+    }
+
+    private Response emptyBodyResponse(int httpStatus) {
+        return Response.status(httpStatus).entity(new Object()).build();
```

Comment:
you can extract new Object() into constant string and name it EMPTY_RESPONSE_BODY